dds/DCPS/XTypes/TypeObject.cpp

```diff
@@ -997,24 +1012,32 @@ namespace {
 
   void compute_dependencies_i(const TypeMap& type_map,
                               const PlainMapLTypeDefn& type,
-                              OPENDDS_SET(TypeIdentifier)& dependencies)
+                              TypeIdentifierSet& dependencies)
   {
     compute_dependencies_i(type_map, type.header, dependencies);
     compute_dependencies_i(type_map, *type.element_identifier, dependencies);
     compute_dependencies_i(type_map, *type.key_identifier, dependencies);
   }
 
-  void compute_dependencies_i(const TypeMap&,
-                              const StronglyConnectedComponentId&,
-                              OPENDDS_SET(TypeIdentifier)&)
+  void compute_dependencies_i(const TypeMap& type_map,
+                              const StronglyConnectedComponentId& type,
+                              TypeIdentifierSet& dependencies)
   {
-    // Do nothing.
+    // An SCCId (identifies the entire component.)
```

Comment:
If I have a topic type T which (eventually) depends on an SCC, and I request the dependencies of just type T... doesn't this code mean I'll _always_ get all the dependencies for the whole SCC (even if I haven't asked explicitly asked for it yet?). Is that what's intended / desired?

Reply:
That is the intended behavior.  If you ask for the dependencies of T, you should get all of the dependencies of T.